### PR TITLE
DOC: update TradingAlgorithm docstring

### PR DESCRIPTION
**Pull request by llllllllll**

### CODE REVIEW

File: zipline/algorithm.py [hunk 1]
```diff
@@ -110,73 +110,78 @@
...
-    my_algo = TradingAlgorithm(initialize, handle_data)
-    stats = my_algo.run(data)
-
+    """A class that represents a trading strategy and parameters to execute
+    the strategy.
+
+    Parameters
+    ----------
+    *args
+        Forwarded ``initialize``.
```
Comment (richafrank) [inline zipline/algorithm.py]: Forwarded "to"

File: zipline/algorithm.py [hunk 2]
```diff
@@ -110,73 +110,78 @@
...
+    initialize : callable[context -> None, optional
```
Comment (richafrank) [inline zipline/algorithm.py]: missing close bracket?

File: zipline/algorithm.py [hunk 3]
```diff
@@ -110,73 +110,78 @@
...
+        Function called on every bar. This is where most logic should be
+        implemented.
+    before_trading_start : callable[(context, data) -> None], optional
+        Function that is called before any bars have been processed each
+        day.
+    analyze : callable[(context, DataFrame) -> None], optional
+        Function that is called at the end of the backtest. This is passed
+        the context and the performance results for the backtest.
+    script : str, optional
+        Algoscript that contains the definitions for the four API
```
Comment (richafrank) [inline zipline/algorithm.py]: I wonder if "API function" is confusing, given `zipline.api`.  Hmm, what about "algo lifecycle functions?"

File: zipline/algorithm.py [hunk 4]
```diff
@@ -110,73 +110,78 @@
...
+        functions and any supporting code.
+    namespace : dict, optional
+        The namespace to execute the algoscript in. By default this is an
+        empty namespace that will include only python built ins.
+    algo_filename : str, optional
+        The filename for the algoscript. This will be used in exception
+        tracebacks. default: '<string>'.
+    data_frequency : {'daily', 'minute'}, optional
```
Comment (richafrank) [inline zipline/algorithm.py]: Should we specify the default?

File: zipline/algorithm.py [hunk 5]
```diff
@@ -110,73 +110,78 @@
...
+        This function will be passed the data for the bar and should
+        return the actual context manager that will be entered.
+    history_container_class : type, optional
+        The type of history container to use. default: HistoryContainer
+    platform : str, optional
+        The platform the simulation is running on. This can be queried for
+        in the simulation with ``get_environment``. This allows algorithms
+        to conditionally execute code based on platform it is running on.
+        default: 'zipline'
+    **kwargs
```
Comment (richafrank) [inline zipline/algorithm.py]: Worth keeping `*args` and `**kwargs` together?

File: zipline/algorithm.py (hunk 4 above)
Comment (llllllllll) [inline zipline/algorithm.py]: the numpy convention for an optional set of arguments is that the first element is the default

File: zipline/algorithm.py (hunk 5 above)
Comment (llllllllll) [inline zipline/algorithm.py]: yeah, I will move this to the front with *args

File: zipline/algorithm.py (hunk 3 above)
Comment (llllllllll) [inline zipline/algorithm.py]: I like that name much better

File: zipline/algorithm.py (hunk 4 above)
Comment (richafrank) [inline zipline/algorithm.py]: Oh ok. TIL

**Pull request merged**